### PR TITLE
Update CI Workflows

### PR DESCRIPTION
**Pull request by embano1**
- [x] Simplify unit tests using matrix
- [x] Drop go:build tags on ruler tests as they seem unneeded now

Signed-off-by: Michael Gasch <15986659+embano1@users.noreply.github.com>

### CODE REVIEW

**Comment by embano1** [comment]
TODO:

- [ ] Update branch protection rules to use the updated unit test name

**Review by timbray — DISMISSED** [review]

File: .github/workflows/go-unit-tests.yaml
Comment (timbray) [inline .github/workflows/go-unit-tests.yaml:23]: I renamed the Ruler test funds, checked these flags, and they seem to work as expected. Now we find out if GitHub can run them.  I considered pulling this fancy matrix into my rename-funcs PR but decided against it because if there's some stupid little syntax problem or other bug you'd be better equipped to fix it.

**Comment by embano1** [comment]
@timbray rebased onto #179 and did three runs, all successful: https://github.com/timbray/quamina/actions/runs/3973740675/usage

**Comment by embano1** [comment]
Update: I made some more modifications, trying to run the `TestRuler` tests together with the regular unit tests. To my surprise, it works. See both runs: https://github.com/timbray/quamina/actions/runs/3974871446

We still have to run `cover` separately due to issues with the `race` detector it seems (tests still time out/get terminated).
But even in the `cover` target, I run `TestRuler` tests (-tags=fromRuler) and they work.

Perhaps we don't need the build tags anymore then?

(disclosure: I updated CI to use Go 1.19, which might have a positive effect here too)

**Comment by timbray** [comment]
Um, if you're running them all together and now it works, that *has* to be go 1.19, right? Unless something else changed?

But if you're convinced that the build tags are no longer necessary, feel free to pull them out as part of this  PR.

Hmm, it's now 9:19AM Pacific time and it looks like the unit tests are somehow blocked from finishing? Looks like 3 hours since last commit?

**Comment by timbray** [comment]
It also occurs to me that I would like to steal a lot more tests from Ruler in the future. Maybe that is an argument for giving them their own spot in the matrix?

**Comment by embano1** [comment]
> Um, if you're running them all together and now it works, that _has_ to be go 1.19, right? Unless something else changed?

It worked with v1.18 too (just used latest in last commit). Not sure what else changed, happy it works :)

> But if you're convinced that the build tags are no longer necessary, feel free to pull them out as part of this PR.

Yup.

> Hmm, it's now 9:19AM Pacific time and it looks like the unit tests are somehow blocked from finishing? Looks like 3 hours since last commit?

Please change this in branch protection rules (new workflow run name). Might have to wait until this is merged.

**Comment by embano1** [comment]
> It also occurs to me that I would like to steal a lot more tests from Ruler in the future. Maybe that is an argument for giving them their own spot in the matrix?

Not sure that's an argument tbh.

**Review by timbray — COMMENTED** [review]
If you're going to remove the build tags, maybe flip the func names back in those two files? Along the lines of

s/^func TestRuler/func Test/

Or I could just roll back #178 since it seems to be useless now.

**Comment by embano1** [comment]
> If you're going to remove the build tags, maybe flip the func names back in those two files? Along the lines of
> 
> s/^func TestRuler/func Test/
> 
> Or I could just roll back #179 since it seems to be useless now.

Actually I like the naming as it indicates `Ruler`-related tests, no?

**Comment by timbray** [comment]
> Actually I like the naming as it indicates `Ruler`-related tests, no?

OK, not fussy.  So, LGTM. I wonder why the problem went away?

**Review by timbray — APPROVED** [review]
Assuming you know how to fix up the branch protection so the unit test completes properly.

**Comment by timbray** [comment]
Oh, still seeing a not-signed problem :(

**Comment by embano1** [comment]
> Oh, still seeing a not-signed problem :(

Haven't added ssh signing. Shall I or merge?

**Comment by timbray** [comment]
> > Oh, still seeing a not-signed problem :(
> 
> Haven't added ssh signing. Shall I or merge?

Go ahead. But let's please set up signing post-1.0?

**Comment by timbray** [comment]
> > > Oh, still seeing a not-signed problem :(
> > 
> > 
> > Haven't added ssh signing. Shall I or merge?
> 
> Go ahead. But let's please set up signing post-1.0?

Go ahead and merge, I meant.

**Pull request merged**
